### DISCUSSION TITLE
How to get flash message from useSubmission to progressively enhance a form with javascript disabled ?

### DISCUSSION THREAD

**Discussion by caguy** [solidjs/solid-router#447] [Q&A] [answered]
Hi everyone,

I'm trying to build a form with Solid Start that would work with Javascript disabled and be progressively enhanced if javascript is disabled.

I'm using Solid Router's server only `action` and `useSubmission` to achieve that, according to [the doc](https://docs.solidjs.com/solid-router/concepts/actions). 

This works great as long as Javascript is enabled, but as soon as I disable it, the useSubmission's return value is empty. I reproduced this behavior in a [minimal sandbox](https://codesandbox.io/p/devbox/agitated-mountain-qt8fkw?file=%2Fsrc%2Fcomponents%2FForm.tsx%3A13%2C1).

I'm not sure if this is a bug or a misuse if the tool : when I inspect the network tab, I can see the flash message returned from the server action being sent as a cookie on the page reload response, but I don't know how to use these values in the render of the page server-side.

I would have expected useSubmission be usable server-side when browser's javascript is disabled, similary to Next's useFormState, but I may just be using it wrong.

**Comment by ryansolid** [comment, marked as answer]
We introduced an error in 0.13.4 of the router that broke this functionality. It will be back in 0.13.6 which i should be releasing today.

**Reply by caguy** [reply]
Thanks, I confirm this works as expected in 0.13.6. 
Solid Start's server actions implementation are really smooth and efficient, very promising!
